### ISSUE TITLE
scylla-sstable: allow providing sstables located in S3 as arguments to all scylla-sstable commands

### ISSUE THREAD

**Issue by denesb** [scylladb/scylladb#20535]
Allow all existing and future commands to seamlessly operate on either local sstables, or sstables located in S3.
The minimum requirement is that a command has to be able to operate on either all-local or all-S3 files. If it is easy to implement, we can allow for mixed arguments (some sstables are local, some are in S3).

**Comment by regevran** [comment]
The following is the relevant parts of the output of `scylla sstable --help` :

```
* decompress: Decompress sstable(s)
* dump-compression-info: Dump content of sstable compression info(s)
* dump-data: Dump content of sstable(s)
* dump-index: Dump content of sstable index(es)
* dump-scylla-metadata: Dump content of sstable scylla metadata(s)
* dump-statistics: Dump content of sstable statistics(s)
* dump-summary: Dump content of sstable summary(es)
* script: Run a script on content of an sstable
* scrub: Scrub the sstable(s), in the specified mode
* shard-of: Print out the shard which 'owns' the sstable
* validate: Validate the sstable(s), same as scrub in validate mode
* validate-checksums: Validate the checksums of the sstable(s)
* write: Write an sstable
* writetime-histogram: Generate a histogram of all the timestamps (writetime)
```

We should make each one of them an separate issue, taking into consideration the ability to share code (e.g. where to read the SStable from).

**Comment by denesb** [comment]
> We should make each one of them an separate issue, taking into consideration the ability to share code (e.g. where to read the SStable from).

I don't think this makes sense. The tool uses a single function to open input sstables: https://github.com/scylladb/scylladb/blob/f3f956841fc4fb720626de2d036478d64c76dd3a/tools/scylla-sstable.cc#L311

This is shared between all commands that have input sstables. So updating this method, will automatically add S3 support to all commands which works with input sstables.

**Comment by regevran** [comment]
It seems that for SSTable that are backed up in S3 `shard-of` and `script` might be meaningless. 
The one that pickup this issue will need to make sure that it make sense to solve-them-all-together and if there is a need it will be split into a few issues.

**Comment by denesb** [comment]
> It seems that for SSTable that are backed up in S3 `shard-of` and `script` might be meaningless. The one that pickup this issue will need to make sure that it make sense to solve-them-all-together and if there is a need it will be split into a few issues.

Why would those commands be meaningless?
`shard-of` is useful to determine that given the current configuration, which shard would own a given sstable, wherever that sstable might be located.
`script` has proven useful several times when we had to create arbitrary stats from the content of sstables, to investigate why ScyllaDB did things a certain way.
